internal/ls/userpreferences.go

```diff
@@ -199,7 +309,35 @@ const (
 	QuotePreferenceSingle  QuotePreference = "single"
 )
 
-func (p *UserPreferences) Parse(config map[string]interface{}) {
+func parseQuotePreference(val any) QuotePreference {
+	if s, ok := val.(string); ok {
+		switch strings.ToLower(s) {
+		case "auto":
+			return QuotePreferenceAuto
+		case "double":
+			return QuotePreferenceDouble
+		case "single":
+			return QuotePreferenceSingle
+		}
+	}
+	return QuotePreferenceUnknown
+}
+
+func (p *UserPreferences) Copy() *UserPreferences {
+	if p == nil {
+		return nil
+	}
+	prefCopy := *p
+	copy(prefCopy.AutoImportSpecifierExcludeRegexes, p.AutoImportSpecifierExcludeRegexes)
+	copy(prefCopy.AutoImportFileExcludePatterns, p.AutoImportFileExcludePatterns)
+	return &prefCopy
+}
+
+func (p *UserPreferences) CopyOrDefault() *UserPreferences {
+	if p == nil {
+		return NewDefaultUserPreferences()
+	}
+	return p.Copy()
 }
```

Comment:
I think we should try to make sure user prefs are always non-nil, and just have the session's initialized default user pref be the default pref, rather than having this method called often.